src/main/java/com/google/sps/utility/FreeTimeUtility.java

```diff
@@ -0,0 +1,142 @@
+  private List<Long> personalHoursPerDay;
+  private List<DatePair> morningFreeInterval;
+  private List<DatePair> workFreeInterval;
+  private List<DatePair> eveningFreeInterval;
+  private static final long HOUR_MILLI = TimeUnit.HOURS.toMillis(1);
+  private static final long DAY_MILLI = TimeUnit.DAYS.toMillis(1);
+  private static final long PERSONAL_BEGIN = 7 * HOUR_MILLI;
+  private static final long WORK_BEGIN = 10 * HOUR_MILLI;
+  private static final long WORK_END = 18 * HOUR_MILLI;
+  private static final long PERSONAL_END = 23 * HOUR_MILLI;
+  private static final int OFFSET = 7;
+
+  /**
+   * Initialize the class with the start day. The work hours are harrrd-coded between 10 AM and 6
+   * PM. The rest of the free hours are between 7 AM and 11 PM.
+   *
+   * @param startTime parameter that gives the time of the start/now.
+   */
+  public FreeTimeUtility(Date startDate) {
+    this.startDate = startDate;
+    Date basisDate = new Date(startDate.getYear(), startDate.getMonth(), startDate.getDate(), 0, 0);
+
+    this.morningFreeInterval = new ArrayList<>();
+    this.workFreeInterval = new ArrayList<>();
+    this.eveningFreeInterval = new ArrayList<>();
+
+    for (int day = 0; day < 5; day++) {
+      Date workStart = new Date(basisDate.getTime() + day * DAY_MILLI + WORK_BEGIN);
+      Date workEnd = new Date(basisDate.getTime() + day * DAY_MILLI + WORK_END);
+      Date personalStart = new Date(basisDate.getTime() + day * DAY_MILLI + PERSONAL_BEGIN);
+      Date personalEnd = new Date(basisDate.getTime() + day * DAY_MILLI + PERSONAL_END);
+      morningFreeInterval.add(new DatePair(personalStart, workStart));
+      workFreeInterval.add(new DatePair(workStart, workEnd));
+      eveningFreeInterval.add(new DatePair(workEnd, personalEnd));
+    }
+    addEvent(basisDate, startDate);
+  }
+
+  /**
+   * Change the free intervals based on the new event added.
+   *
+   * @param evenStart parameter that gives start time.
+   * @param eventEnd parameter that gives end tome of the event
+   */
+  public void addEvent(Date eventStart, Date eventEnd) {
+    this.morningFreeInterval = updateInterval(this.morningFreeInterval, eventStart, eventEnd);
+    this.workFreeInterval = updateInterval(this.workFreeInterval, eventStart, eventEnd);
+    this.eveningFreeInterval = updateInterval(this.eveningFreeInterval, eventStart, eventEnd);
+  }
+
+  /**
+   * Method to update the free intervals in a particular list of intervals.
+   *
+   * @param freeTime the list of intervals to update.
+   * @param eventStart the start time of the new event
+   * @param eventEnd the end time of the new event
+   * @return the new list of intervals
+   */
+  private List<DatePair> updateInterval(List<DatePair> freeTime, Date eventStart, Date eventEnd) {
+    List<DatePair> updatedTime = new ArrayList<>();
+    for (DatePair interval : freeTime) {
+      if (interval.getKey().before(eventStart) && interval.getValue().after(eventEnd)) {
```

Comment:
Add a brief comment documenting each case here. Also document what the unwritten "else" case is.

Reply:
`Resolved`